consensus/safety-rules/src/safety_rules.rs

```diff
@@ -233,16 +245,11 @@ impl TSafetyRules for SafetyRules {
                 last_voted_round: self.persistent_storage.last_voted_round()?,
             });
         }
+        proposed_block.validate_signatures(self.verifier()?)?;
 
-        let preferred_round = self.persistent_storage.preferred_round()?;
-        if proposed_block.quorum_cert().certified_block().round() < preferred_round {
-            debug!(
-                "Vote proposal certified round is lower than preferred round, {} < {}",
-                proposed_block.quorum_cert().certified_block().round(),
-                preferred_round,
-            );
-            return Err(Error::ProposalRoundLowerThenPreferredBlock { preferred_round });
-        }
+        let qc = proposed_block.quorum_cert();
+        self.verify_qc(qc)?;
+        self.verify_qc_proposed_round(qc)?;
```

Comment:
I don't think there should be any issue from merging these together. In fact, we're planning on merging update into construct_and_sign_vote and sign_proposal to simplify the API. cc: @zekun000 

if you wanna add a commit to this stick to remove update and adjust those two calls, that would be great.

Reply:
Yes, and I'm actually a little confused about the second check. In the first check we made sure that the parent's round is at least the preferred round, and in the second check it checks that the proposed round is at least the preferred round. But wouldn't the proposed round be 1 larger than the parent round?

Reply:
So that makes it pretty easy to merge right?

* QuorumCert::verify should ensure that the parent round < certified round, which it currently does not
* SafetyRules::verify_qc should ensure that parent round >= preferred round (which it currently does)

Reply:
`SafetyRules::verify_qc` should only verify the signatures if we want to use it both in `update` and `vote` because
- `update` requires `parent_round > preferred_round` - this is 2-chain
- `vote` requires `certified_round > preferred_round` - this is 1-chain

Reply:
Which signature is `SafetyRules::verify_qc` verifying?